File: src/core/damage.py
from dataclasses import dataclass
from typing import *
from constant_types import DamageType

# RAW UNMITIGATED DAMAGE INFORMATION
@dataclass
class Damage:
    value: float
    frame_number: int
    dmg_type: Optional[DamageType] = None
    source_unit_id: Optional[int] = None
    target_unit_id: Optional[int] = None
    spell_name: Optional[str] = None
    crit: bool = False
    dot: bool = False
    heal: bool = False
    match_id: Optional[int] = None
    

File: csharp/MicroAutoChess.Core/Core/Damage.cs
namespace MicroAutoChess.Core
{
    public class Damage
    {
        public double Value { get; set; }
        public int FrameNumber { get; set; }
        public DamageType? DmgType { get; set; }
        public int? SourceUnitId { get; set; }
        public int? TargetUnitId { get; set; }
        public string? SpellName { get; set; }
        public bool Crit { get; set; } = false;
        public bool Dot { get; set; } = false;
        public bool Heal { get; set; } = false;
        public int? MatchId { get; set; }

        public Damage() { }

        public Damage(double value, int frameNumber)
        {
            Value = value;
            FrameNumber = frameNumber;
        }
    }
}
